### PR TITLE
CRM_Utils_Check - Generate a `notice` if MySQL doesn't support accents

### PR DESCRIPTION
**Pull request by totten**
Overview
----------------------------------------
Generate a `notice` if MySQL doesn't support storage of accents

Before
----------------------------------------
No warnings or notices.

After
----------------------------------------
The status-check generates a `notice`

screen shot 2018-04-30 at 7 54 18 pm

Comments
----------------------------------------
This is a follow-up to #12060.  I'm submitting it for discussion -- but I'm not
actually confident in the change, so please don't merge unless you affirmatively
agree.  Let me recount the context.

PR #12060 aimed to address an issue where `testInternationalStrings()` failed on
one system (`test-ubu1604-1`) but succeeded another (`test-ubu1204-5`).  The
content of `character_set_database` was suggested as a distinguishing
characteristic, and #12060 basically says, "Don't test if we don't expect it to work."
That's nice for developers because it reduces the noisiness of the test suite.

But it also reduces the impact of the `testInternationalStrings()` -- because a
non-working environment is considered passing.  Is it legit to say "it's not the
devteam's problem if some environments don't work"?  Yes, that's legit.  But if
we have to kick the ball over the sysadmin, then let's finish job and give them
a notice so they can fix the environment.

Both #12060 and this patch are premised on the correctness of
`supportStorageOfAccents()`.  I'm not really confident in it, though.  On my local
system, `supportStorageOfAccents()` returns `FALSE`, but I'm able to create and view
a contact with last name `Iñtërnâtiônàlizætiøn` -- via either web UI or CLI
(`cv api contact.create`). (Note: CLI displays JSON by default, and JSON converts to `\uXXX` notation. It's more readable with `--out=pretty`) That casts some doubt on `supportStorageOfAccents()`, but it's not decisive. (The check may be important for some other reason/use-case).

In short -- if `supportStorageOfAccents()` is working correctly, then we should
merge this and notify site-admins.  If `supportStorageOfAccents()` is not working
correctly, then we probably need something else to fix #12060.

Note: The hyperlink currently points to a wiki page (same ideas the notices in `CRM_Utils_Component_Security`) which could be implemented or redirected to docs when someone has steps to resolve. I don't have those steps (and I don't even know if the test is right!) -- but I'm submitting this PR because I think we should be logically consistent about the matter.

### CODE REVIEW

**Comment by colemanw** [comment]
@totten I was about to merge this until I read about your problem with `supportStorageOfAccents()`. So are you saying that your local system *is* UTF-8 and *is* working correctly to store international characters, but with this branch checked out it will give you a false-positive system warning?

**Comment by totten** [comment]
>  it will give you a false-positive system warning?

Probably... at least, when I tested locally, it seemed like a false-negative. But in another context (`api_v3_ContactTest::testInternationalStrings()`), `supportStorageOfAccents()` does discriminate between the OK environment of `test-ubu1204-5` and the failing environment on `test-ubu1604-1`. Which means either:

1. `supportStorageOfAccents()` yields the right outcome on the test boxes but for the wrong reason.
2. `supportStorageOfAccents()` yields the right outcome for the right reason, but my local, manual testing didn't touch on the real problem.

I'm pretty sure it's `#1` -- Seamus led some discussion/investigation on `infrastructure` on what differentiates the test boxes, and it's a really tough nut. (Finding *one* discriminator took a lot of careful work... finding an another discriminator is even more work.) The question is whether we dig-in on making the function accurate or cut our losses and move on.

**Comment by seamuslee001** [comment]
Yeh i'm in the `#1` camp as well now, Its excluding the test on the current main PR test box which was working but it is excluding it rightly from the 1604-1 test box. There appears i think to be some dark magic in how `locale` is interacting with MySQL because it feels like the data isn't getting stored in a utf8 compatible way.

**Comment by eileenmcnaughton** [comment]
@totten I also agree that 1 is the solution - given that - does this check still confer some value or not?

**Comment by eileenmcnaughton** [comment]
@totten should we merge or ditch this?

**Comment by totten** [comment]
Let's ditch this PR. Instead, we should put some kind of comment/warning in the docblock for `supportStorageOfAccents()` so that people understand the context/purpose (i.e. specific problem in running tests) and don't try to use it for other things.
